From c80dd230adf23b90c7ca40ca1777c1fc236bf486 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Thu, 1 Feb 2018 17:01:43 +0000
Subject: [PATCH 13/17] LiveDisplayTile: Avoid NPE during boot up phase

 * During boot, with LineageHW not yet initialized there is no way to
   check the current LiveDisplay mode. In such cases, fallback to AUTO.

Change-Id: Ifc004c86dfd68f73b8a362a1c833cbc3e7397811
---
 .../src/com/android/systemui/qs/tiles/LiveDisplayTile.java     | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/LiveDisplayTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/LiveDisplayTile.java
index 780487f..9c5c6754 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/LiveDisplayTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/LiveDisplayTile.java
@@ -18,6 +18,7 @@
 package com.android.systemui.qs.tiles;
 
 import static lineageos.hardware.LiveDisplayManager.FEATURE_MANAGED_OUTDOOR_MODE;
+import static lineageos.hardware.LiveDisplayManager.MODE_AUTO;
 import static lineageos.hardware.LiveDisplayManager.MODE_DAY;
 import static lineageos.hardware.LiveDisplayManager.MODE_OUTDOOR;
 
@@ -152,7 +153,14 @@ public class LiveDisplayTile extends QSTileImpl<LiveDisplayState> {
     }
 
     private int getCurrentModeIndex() {
-        return ArrayUtils.indexOf(mValues, String.valueOf(mLiveDisplay.getMode()));
+        String currentLiveDisplayMode = null;
+        try {
+            currentLiveDisplayMode = String.valueOf(mLiveDisplay.getMode());
+        } catch (NullPointerException e) {
+            currentLiveDisplayMode = String.valueOf(MODE_AUTO);
+        } finally {
+            return ArrayUtils.indexOf(mValues, currentLiveDisplayMode);
+        }
     }
 
     private void changeToNextMode() {
-- 
2.7.4

